File: src/main/java/pl/khuzzuk/wfrp/helper/model/knowledge/magic/SpellSchoolLevel.java
package pl.khuzzuk.wfrp.helper.model.knowledge.magic;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.envers.Audited;
import org.hibernate.envers.RelationTargetAuditMode;
import pl.javahello.DTO;
import pl.khuzzuk.wfrp.helper.repo.BaseEntity;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Getter
@Setter
@Audited
@Entity
@Table(schema = "creature", name = "person_spell_school_level")
@DTO
public class SpellSchoolLevel extends BaseEntity {
    @ManyToOne
    @JoinColumn(name = "spell_school_id")
    @Audited(targetAuditMode = RelationTargetAuditMode.NOT_AUDITED)
    private SpellSchool spellSchool;
    private int level;
}

File: src/wfrp-helper-web/src/model/knowledge/SpellSchoolLevel.js
import Entity from "../Entity";
import SpellSchool from "./SpellSchool";

export default class SpellSchoolLevel extends Entity {
    spellSchool: SpellSchool;
    level: number = 0;


    updateWith(entity: *): SpellSchoolLevel {
        super.updateWith(entity);
        this.updateEntityProp(entity, 'spellSchool', () => new SpellSchool());
        return this;
    }
}